extensions/stackdriver/edges/edge_reporter.cc

```diff
@@ -40,25 +40,39 @@ using google::cloud::meshtelemetry::v1alpha1::
     TrafficAssertion_Protocol_PROTOCOL_TCP;
 using google::cloud::meshtelemetry::v1alpha1::WorkloadInstance;
 
+constexpr char kUnknown[] = "unknown";
+
+std::string valueOrUnknown(std::string value) {
```

Comment:
avoid copy of string: maybe use string_view as return type and receive a `const std::string&`?

Reply:
updated to use `const std::string&`. switching to string_view will require more code munging as they aren't accepted in the proto setters.

Reply:
still copies to return. but it's one less copy.

Reply:
shouldn't modern c++ compilers be able to avoid copies in this scenario? i find the standard impossible to parse, but this feels like a situation in which copy eliding could apply.

Reply:
there are too many compilers out there to say definitively. probably the very latest ones do, but not sure about older clang, gcc, and emscripten.